modules/images/webp-uploads/load.php

```diff
@@ -493,6 +477,29 @@ function webp_uploads_img_tag_update_mime_type( $image, $context, $attachment_id
 			continue;
 		}
 
```

Comment:
Commenting here since GitHub doesn't let me comment on lines 455-464: That is where the full image is replaced, so we should add a similar file size comparison for that as well and only do the replacement if needed. Basically compare `$metadata['sources'][ $target_mime ]['filesize']` with `$metadata['sources'][ $original_mime_type ]['filesize']`.

This also should only happen based on the filter though, so this relates to my other comment. We should only call this filter once somewhere in the beginning of this function and then check the value just by checking the value stored in a variable.

Reply:
Thanks @felixarntz I've updated the PR to apply the logic to the full image as well